mindmeld/openapi/custom_action.yaml

```diff
@@ -0,0 +1,187 @@
+      entities:
+        type: "array"
+        description: "A list of entities in the current query."
+        items:
+          type: object
+          $ref: "#/definitions/Entity"
+      context:
+        type: object
+        description: "Map containing front-end client state that is passed to the application from the client in the request."
+      params:
+        type: object
+        $ref: "#/definitions/Params"
+        description: "Map of stored data across multiple dialogue turns."
+  Transcript:
+    type: "object"
+    properties:
+      transcript:
+        type: "string"
+        description: "Transcription"
+      confidence:
+        type: "number"
+        description: "Transcription confidence"
+  Directive:
+    type: "object"
+    properties:
+      name:
+        type: "string"
+        description: "Directive Name"
+      type:
+        type: "string"
+        description: "Directive Type"
+        enum:
+        - "view"
+        - "action"
+      payload:
+        type: "object"
+        description: "JSON payload"
+  Domain:
+    type: object
+    properties:
+      value:
+        type: "string"
+      confidence:
+        type: "number"
+  Intent:
+    type: object
+    properties:
+      value:
+        type: "string"
+      confidence:
+        type: "number"
+  Entity:
+    type: "object"
+    properties:
+      text:
+        type: "string"
+        description: "The text contents that span the entity"
+      type:
+        type: "string"
+        description: "Entity type"
+      role:
+        type: "string"
+        description: "Role type"
+      value:
+        type: "object"
+        description: "The resolved value of the entity"
+      display_text:
+        type: string
+        description: "A human readable text representation of the entity for use in natural language responses."
+      confidence:
+        type: number
+        description: "A confidence value from 0 to 1 about how confident the entity recognizer was for the given class label."
+      is_system_entity:
+        type: boolean
+        description: "True if the entity is a system entity"
+  Params:
+    type: object
+    properties:
+      target_dialogue_state:
+        type: string
+        description: "The name of next turn's dialogue state (if set)"
+      dynamic_resoure:
+        type: string
+        description: "The additional values for gazetteer"
+      allowed_intents:
+        type: array
+        items:
+          type: string
+          description: "The name of the intent"
+        description: "The list of allowed intents for next turn (if set)"
+      time_zone:
+        type: string
+        description: "The time zone of the request"
+      language:
+        type: string
+        description: "The language of the request"
+      locale:
+        type: string
+        description: "The locale of the request"
+      timestamp:
+        type: string
+        description: "The timestamp of the request"
+
+  Responder:
```

Comment:
Can we please document these fields based on the descriptions listed here: https://www.mindmeld.com/docs/userguide/dm.html?highlight=directive#responder. The current documentation does not explain to the developer what the objects do.

Reply:
This is the documentation for the API, not the documentation for the Responder so I am not sure if we should include all the descriptions from the Responder documentation. I think we should include the link for the documentation and that might be enough.